File: api/Entities/Campaign.cs
using api.Entities.Rulesets;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Entities
{
    public class Campaign
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }


        public int RulesetID { get; set; }
        [ForeignKey("RulesetID")]
        public Ruleset Ruleset { get; set; }

        public string AppUserId { get; set; }
        [ForeignKey("AppUserId")]
        public AppUser User { get; set; }
    }
}

File: src/app/campaign/models/campaign.ts
import { Ruleset } from "../../models/core/rulesets/ruleset";
import { Spell } from "../modules/spell/models/spell";
import IFormEditable from "src/app/models/core/app/forms/iFormEditable";

export class Campaign implements IFormEditable {
	public id: number;
	public name: string;
	public description: number;
	public ruleset: Ruleset;
	public user: string;

	public monsters: any[];
	public spells: Spell[];

	public static getDefault(): Campaign {
		const campaign = new Campaign();
		campaign.id = -1;
		campaign.user = undefined;

		campaign.ruleset = Ruleset.getDefault();
		campaign.monsters = [];
		campaign.spells = [];

		return campaign;
	}

	public static isEqual(a: Campaign, b: Campaign): boolean {
		return (
			a.id === b.id &&
			a.name === b.name &&
			a.description === b.description
		);
	}
}
